internal/advertisement-operator/controller.go

```diff
@@ -140,12 +140,26 @@ func (r *AdvertisementReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // set Advertisement reference in related ForeignCluster
 func (r *AdvertisementReconciler) UpdateForeignCluster(adv *protocolv1.Advertisement) (error, bool) {
-	tmp, err := r.DiscoveryClient.Resource("foreignclusters").Get(adv.Spec.ClusterId, metav1.GetOptions{})
+	tmp, err := r.DiscoveryClient.Resource("foreignclusters").List(metav1.ListOptions{
```

Comment:
why do you use `List` if you're interested only on 1 ForeignCluster object?

Reply:
Because I can use LabelSelector. When a foreign cluster is created is added a label indicating its clusterId, so I can retrieve it by this label that is in our control. If someone creates a ForeignClsuter manually it's possible that the name and the clusterId are different